### PR TITLE
Manual stage trigger on dashboard

### PR DESCRIPTION
**Pull request by GaneshSPatil**
Issue: #7687

#### Done:
- [x] ~~Bump up dashboard api~~ Dashboard API has already been bumped up from v3 to v4 to add empty pipeline group information. 
- [x] Add approval_type information to the dashboard api.
- [x] Add manual gate button to the manual stages.
- [x] Disable manual gate when:
  - [x] the previous stage hasn't run or still is in progress.
  - [x] the current stage has already been scheduled.
  - [x] user does not have permissions to operate on the pipeline.
- [x] Add help text.
- [x] Bind stage trigger api responses to the pipeline tile.
- [x] Disable manual gate when stage permissions are overridden. Dashboard API does not include permissions at individual stage.
- [x] Disable manual trigger when allow_only_on_success_of_previous_stage is set to true and previous stage is either failed or cancelled.

#### Pending:
- [ ] ~~Support force scheduling next stage when previous stage does not complete successfully (failed/cancelled).~~ This option will be considered as part of stage overview popup.

### Screenshots:

Screen Shot 2020-07-01 at 2 18 28 PM

Screen Shot 2020-07-01 at 2 17 54 PM

Screen Shot 2020-07-01 at 2 25 19 PM

The error scenario screenshot can be ignored. When permissions are implemented, the manual gate icon will be disabled when users don't permission to schedule the stage which avoids this error.
It is added to demonstrate error binding with the pipeline tile.

Screen Shot 2020-07-01 at 2 24 48 PM

### CODE REVIEW

File: server/src/main/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts [hunk 1]
```diff
@@ -258,6 +258,10 @@ export class SparkRoutes {
     return `/go/api/admin/maintenance_mode/info`;
   }
 
+  static stageTriggerPath(pipelineName: string, pipelineCounter: number, stageName: string): string {
+    return `/go/api/stages/${pipelineName}/${pipelineCounter}/${stageName}/run`;
+  }
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts:263]: Duplicate method: `runStage`

**Comment by maheshp** [comment]
@GaneshSPatil we will have to disable the stage trigger when previous stage has failed and the pipeline is configured to allow stage trigger only on success of previous stage. (

**Comment by GaneshSPatil** [comment]
To be merged for `20.7.0`

**Comment by GaneshSPatil** [comment]
@maheshp @arvindsv -- merging. Some of the work done as part of this PR is required for stage overview.
Any review feedback on this PR will be addressed separately.

**Comment by kritika-singh3** [comment]
Can/Should we make the message consistent for both the pages?

Dashboard:
Screen Shot 2020-08-10 at 10 50 47 AM

Pipeline Activity:
Screen Shot 2020-08-10 at 10 51 07 AM

**Comment by maheshp** [comment]
Verified on GoCD Version: 20.7.0 (12081-b3abf30ba7f72def573ac96246eab1ad764ec184)

**Pull request merged**